### PR TITLE
init project

### PR DESCRIPTION
**Pull request by YuvalLeviMoveoHLS**

### CODE REVIEW

File: src/components/Article/article.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        <Card sx={CardSx}>
+          <CardContent sx={CardContentSx}>
+            <Typography sx={DateAndSourceSharedSx}>{publishedAt}</Typography>
+            <Typography sx={ArticleTitleSx}>{title}</Typography>
+            <Typography sx={DateAndSourceSharedSx}>{source.name}</Typography>
+            <Typography variant="body2" sx={ArticleContentSx}>
+              {content}
+            </Typography>
+            <CardActions sx={CardActionsSx}>
+              <StyledButton onClick={() => handleButtonClick(url)}>
```
Comment (ZivCohenMoveo) [inline src/components/Article/article.tsx]:
```suggestion
              <StyledButton onClick={handleButtonClick}>
```

```
  const handleButtonClick = () => {
      window.open(url);
    };
```

File: src/components/Article/articleTest.tsx
Comment (ZivCohenMoveo) [inline src/components/Article/articleTest.tsx:1]: What is this file for?

File: src/components/Button/styledButton.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// delete stack! no  needed
+type StyledButtonProps = {
+  children: ReactNode;
+  sx?: SxProps;
+  onClick?: () => void;
+};
+
+const StyledButton: FC<StyledButtonProps> = ({ children, sx, onClick }) => {
+  return (
+    <Stack spacing={2} direction="row">
```
Comment (ZivCohenMoveo) [inline src/components/Button/styledButton.tsx]: Why do you need `Stack`?

File: src/components/Button/styledButton.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      <Button
+        sx={{ ...buttonSx, ...sx }}
+        endIcon={
+          <img src={arrowRight} onClick={() => console.log("Clicked")} />
```
Comment (ZivCohenMoveo) [inline src/components/Button/styledButton.tsx]: onClick on the icon?

File: src/components/CoolSelectDropdown/CoolSelectDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    <Box sx={{ minWidth: 120 }}>
+      <FormControl fullWidth>
+        <InputLabel id="demo-simple-select-label"></InputLabel>
+        <Select
+          labelId="demo-simple-select-label"
+          placeholder={placeholder}
+          id="demo-simple-select"
+          value={defaultValue}
+          defaultValue={defaultValue}
+          label="Age"
```
Comment (ZivCohenMoveo) [inline src/components/CoolSelectDropdown/CoolSelectDropdown.tsx]: Should be generic or from outside by prop

File: src/components/FilterDropDown/filterDropDown.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  formControlStyle?: SxProps; // for StoryBook
+  menuItemStyle?: SxProps; // for StoryBook
+};
+
+const FilterDropDown: FC<FilterDropDownProps> = ({
+  selectOptions,
+  defaultValue,
+  placeholder,
+  children,
+  isAppHeader = false,
```
Comment (ZivCohenMoveo) [inline src/components/FilterDropDown/filterDropDown.tsx:39]: What is `isAppHeader`?

File: src/components/FilterDropDown/filterDropDown.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        onChange={handleChange}
+        IconComponent={ArrowDownIcon}
+        inputProps={{ "aria-label": "Value" }}
+        sx={{ ...DropDownStyleSx, ...filterDropDownStyle }}
+        renderValue={(selected) => {
+          if (selected === "") {
+            return <span>{placeholder}</span>;
+          }
+          return selected;
+        }}
```
Comment (ZivCohenMoveo) [inline src/components/FilterDropDown/filterDropDown.tsx]: Extract to function

File: src/components/FilterDropDown/filterDropDown.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    // <Box sx={{ minWidth: 120 }}>
+    <FormControl sx={{ ...FormControlStyleSx, ...formControlStyle }}>
+      <Select
+        MenuProps={MenuPropsStyleSx}
+        displayEmpty
+        value={selectedFilterValue}
+        onChange={handleChange}
+        IconComponent={ArrowDownIcon}
+        inputProps={{ "aria-label": "Value" }}
+        sx={{ ...DropDownStyleSx, ...filterDropDownStyle }}
```
Comment (ZivCohenMoveo) [inline src/components/FilterDropDown/filterDropDown.tsx]: Check if you need all props here

File: src/components/Input/input.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { StyledInput, InputWrapper, StyledSearchIcon } from "./input.style";
+import SearchIcon from "../../assets/svg/searchIcon.svg";
+import FilterDropDown from "../../components/FilterDropDown/filterDropDown";
+import { selectOptionsAppHeader } from "../../mockData/selectOptionsDropDownAppHeader";
+
+export interface SearchInput {
+  placeholder?: string;
+}
+const SearchInput: React.FC<SearchInput> = ({ placeholder = "Search" }) => {
```
Comment (ZivCohenMoveo) [inline src/components/Input/input.tsx]: The name of the component should be like the filename

File: src/components/Input/input.tsx
Comment (ZivCohenMoveo) [inline src/components/Input/input.tsx:1]: Separate between searchInput and Input or use flag for it

File: src/stories/Button.stories.ts
Comment (ZivCohenMoveo) [inline src/stories/Button.stories.ts:1]: Maybe move all stories files to Components folder

File: .vite/deps/@mui_material_FormControl.js
Comment (ZivCohenMoveo) [inline .vite/deps/@mui_material_FormControl.js:1]: I think that .vite/deps should be in gitignore

**Review by ZivCohenMoveo — CHANGES_REQUESTED** [review]
Looks nice!

General comments:
1. Remove all comment out code
2. Organize the files and the folders (stories, components, etc...)

File: src/components/FilterDropDown/filterDropDown.tsx (hunk 5 above)
Comment (YuvalLeviMoveoHLS) [reply src/components/FilterDropDown/filterDropDown.tsx:39]: The site has two different designs for the dropdown. This variable helps me to know which styling to choose.

File: src/components/Input/input.tsx
Comment (YuvalLeviMoveoHLS) [reply src/components/Input/input.tsx:1]: I fixed it at another branch

**Review by ZivCohenMoveo — APPROVED** [review]

**Pull request merged**